homeassistant/components/climate/evohome.py

```diff
@@ -53,156 +65,420 @@
-        if 'timeUntil' in self._status['systemModeStatus']:
-            data['timeUntil'] = self._status['systemModeStatus']['timeUntil']
-        data['activeFaults'] = self._status['activeFaults']
-        return data
+        return self._config['setpointCapabilities']['minHeatSetpoint']
 
     @property
-    def operation_list(self):
-        """Return the list of available operations."""
-        return HA_OP_LIST
+    def max_temp(self):
+        """Return the minimum target temperature of a evohome Zone.
+
+        The default is 35 (in Celsius), but it is configurable within 5-35.
+        """
+        return self._config['setpointCapabilities']['maxHeatSetpoint']
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature of the evohome Zone."""
+        return self._status['setpointStatus']['targetHeatTemperature']
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature of the evohome Zone."""
+        return self._status['temperatureStatus']['temperature']
+
+    @property
+    def current_operation(self):
+        """Return the current operating mode of the evohome Zone.
+
+        The evohome Zones that are in 'FollowSchedule' mode inherit their
+        actual operating mode from the Controller.
+        """
+        evo_data = self.hass.data[DATA_EVOHOME]
+
+        system_mode = evo_data['status']['systemModeStatus']['mode']
+        setpoint_mode = self._status['setpointStatus']['setpointMode']
+
+        if setpoint_mode == EVO_FOLLOW:
+            # then inherit state from the controller
+            if system_mode == EVO_RESET:
+                current_operation = TCS_STATE_TO_HA.get(EVO_AUTO)
+            else:
+                current_operation = TCS_STATE_TO_HA.get(system_mode)
+        else:
+            current_operation = ZONE_STATE_TO_HA.get(setpoint_mode)
+
+        return current_operation
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if the evohome Zone is off.
+
+        A Zone is considered off if its target temp is set to its minimum, and
+        it is not following its schedule (i.e. not in 'FollowSchedule' mode).
+        """
+        is_off = \
+            self.target_temperature == self.min_temp and \
+            self._status['setpointStatus']['setpointMode'] == EVO_PERMOVER
+        return not is_off
+
+    def _set_temperature(self, temperature, until=None):
+        """Set the new target temperature of a Zone.
+
+        temperature is required, until can be:
+          - strftime('%Y-%m-%dT%H:%M:%SZ') for TemporaryOverride, or
+          - None for PermanentOverride (i.e. indefinitely)
+        """
+        try:
+            self._obj.set_temperature(temperature, until)
+        except HTTPError as err:
+            if not self._handle_exception("HTTPError", str(err)):               # noqa: E501; pylint: disable=no-member
+                raise
```

Comment:
Don't raise. Log error and return.

Reply:
This `raise` shouldn't have been there - it's a regression, and has now been removed.